Maple2.Server.DebugGame/Graphics/Scene/ICameraController.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System.Numerics;
+
+namespace Maple2.Server.DebugGame.Graphics.Scene;
+
+/// <summary>
+/// Interface for camera controllers that can manipulate a camera
+/// </summary>
+public interface ICameraController {
+    /// <summary>
+    /// The camera being controlled
+    /// </summary>
+    Camera Camera { get; }
+
+    /// <summary>
+    /// Input state for handling user input
+    /// </summary>
+    InputState InputState { get; }
+
+
+
+    /// <summary>
+    /// Current camera position
+    /// </summary>
+    Vector3 CameraPosition { get; }
+
+    /// <summary>
+    /// Current camera rotation
+    /// </summary>
+    Quaternion CameraRotation { get; }
```

Comment:
Shouldn't use quaternions. I'm on the fence about using these in general because anything you need would be in Camera.Transform